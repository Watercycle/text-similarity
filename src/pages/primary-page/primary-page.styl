@import '~/src/shared-styles/colors'

.root :global
    color: $primary

    .card-panel
        background: red
        width: 20%
        height: 20%
