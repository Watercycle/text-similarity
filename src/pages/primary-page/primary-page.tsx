import { Component } from 'solid-js'
import styles from './primary-page.styl'
import { WcButton } from '../../shared-components/button/button'

export interface Props {}

export const PrimaryPage: Component<Props> = props =>
  <>
    <div class={styles.root}>Hello World!</div>
    <WcButton>Some Text</WcButton>
  </>
