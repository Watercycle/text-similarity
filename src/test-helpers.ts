import * as dom from 'solid-js/dom'

export function render(jsx: JSX.Element): string {
  const div = document.createElement('div')

  const dispose = dom.render(() => jsx, div)
  dispose()

  return div.children[0].outerHTML;
}

export function assertMatchesSnapshot(jsx: JSX.Element) {
  const html = render(jsx);
  expect(html).toMatchSnapshot()
}
