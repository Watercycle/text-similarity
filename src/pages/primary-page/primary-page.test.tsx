import { assertMatchesSnapshot } from "../../test-helpers"
import { PrimaryPage } from './primary-page'

describe('PrimaryPage', () => {
    it('renders', () => {
        assertMatchesSnapshot(() => <PrimaryPage/>)
    })    
})
