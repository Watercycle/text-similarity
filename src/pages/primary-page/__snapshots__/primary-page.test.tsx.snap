// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryPage renders 1`] = `"<div class=\\"root\\">Hello World!</div>"`;
